src/backend/utils/cache/inval.c

```diff
@@ -552,6 +552,30 @@ RegisterCatcacheInvalidation(int cacheId,
 								   cacheId, hashValue, dbId);
 }
 
+/*
+ * RegisterENRCatcacheInvalidation
+ *
+ * Register an invalidation event for an ENR catcache tuple entry. 
+ * 
+ * We save the hashvalue so that it can be crosschecked and excluded from 
+ * the SI queue later.
+ * 
+ * This is necessary because there isn't enough information in a 
+ * Catcache SI message itself to determine whether it's applying to ENR.
+ * 
+ * Saved messages are cleared at the end of the current transaction
+ * (see ClearSavedCatcacheMessages).
+ */
+static void
+RegisterENRCatcacheInvalidation(int cacheId,
```

Comment:
how do we validate that we are locally invalidating cache wherever needed? we should have test for each use case which applies to temp tables catalog tables

Reply:
added tests in extension